lighthouse/lighthouse-auth.test.ts

```diff
@@ -197,33 +193,62 @@ describe('Lighthouse Pre-Login Steps', () => {
     });
   });
 
-  describe('loadInitialPage', () => {
+  describe('checkForInitialLoad', () => {
     it('should load the initial page right away', async () => {
-      process.env.PREVIEW_URL = previewUrl;
       successfulTestMock();
 
-      await loadInitialPage(initialPage, `${previewUrl}/accountLists`, 500);
-      expect(initialPage.goto).toHaveBeenCalledTimes(1);
+      await checkForInitialLoad(initialPage, 0, 500);
+      expect(initialPage.waitForSelector).toHaveBeenCalledTimes(1);
     }, 500);
 
     it('should wait until the page is available', async () => {
       successfulTestMock();
 
-      process.env.PREVIEW_URL = previewUrl;
-      const initialResponse = {
-        status: 404,
-      };
-
-      initialPage.goto
-        .mockImplementationOnce(() => {
-          return Promise.resolve(initialResponse);
+      initialPage.waitForSelector
+        .mockImplementationOnce((selector) => {
+          if (selector === '#__next') {
+            return Promise.reject('Timed out waiting to find element');
+          }
+          return Promise.resolve({});
         })
         .mockImplementationOnce(() => {
-          return Promise.resolve({ status: 200 });
+          return Promise.resolve({});
+        });
+
+      await checkForInitialLoad(initialPage, 0, 500);
+      expect(initialPage.waitForSelector).toHaveBeenCalledTimes(2);
+    });
+
+    it('should stop trying after max tries', async () => {
+      successfulTestMock();
+
+      const promiseWithTimeout = (timeout) => {
+        return new Promise((resolve, reject) => {
+          setTimeout(() => {
+            reject(new Error('Timed out waiting to find element'));
+          }, timeout);
         });
+      };
+
+      const maxTries = 14;
+      const timeout = 100;
+      for (let i = 0; i < maxTries; i++) {
+        initialPage.waitForSelector.mockImplementationOnce(
+          async () => await promiseWithTimeout(timeout),
+        );
+      }
 
-      await loadInitialPage(initialPage, `${previewUrl}/accountLists`, 500);
-      expect(initialPage.goto).toHaveBeenCalledTimes(2);
+      initialPage.waitForSelector.mockImplementationOnce(() => {
+        return Promise.resolve({});
+      });
```

Comment:
This and the one on line 214 could be simplified to use `mockResolvedValueOnce` unless you want to keep it as-is to match the others.

```suggestion
      initialPage.waitForSelector.mockResolvedValueOnce({});
```